pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -502,6 +502,14 @@ func (c *Cluster) getOSDInfo(d *appsv1.Deployment) (OSDInfo, error) {
 		}
 	}
 
+	// if the ROOK_TOPOLOGY_AFFINITY env var was not found in the loop above, detect it from the node
```

Comment:
This `if` block sits in between the `localtionFound` for loop (line 495) and `locationFound` condition.  Suggesting to move it either above the `locationFound` code or below it.